File: examples_dotnet/ExampleFile.cs
using LibZupt;
using System.Text;

namespace ZuptExample;

internal static class ExampleFile
{
    private static string MakeTempDir()
    {
        var dir = Path.Combine(Path.GetTempPath(), "zupt_example_" + Guid.NewGuid().ToString("N"));
        Directory.CreateDirectory(dir);
        return dir;
    }

    public static void Run()
    {
        Console.WriteLine(new string('=', 60));
        Console.WriteLine("libzupt - File Encryption/Decryption Example");
        Console.WriteLine(new string('=', 60));
        Console.WriteLine();

        Console.WriteLine("1. Generating key pair...");
        var keygen = new KeyGenerator();
        var keypair = keygen.GenerateKeyPair();
        Console.WriteLine("   Key pair generated");
        Console.WriteLine();

        var encryptor = new Encryptor(keypair.PublicKey);
        var decryptor = new Decryptor(keypair.SecretKey);

        var tmpDir = MakeTempDir();
        try
        {
            var testFile = Path.Combine(tmpDir, "example.txt");
            var content = new StringBuilder();
            content.AppendLine("This is a secret text file.");
            content.AppendLine("Line 2: Contains sensitive information.");
            content.AppendLine("Line 3: End of file.");
            File.WriteAllText(testFile, content.ToString());
            byte[] originalContent = Encoding.UTF8.GetBytes(content.ToString());

            Console.WriteLine($"2. Created test file: {testFile}");
            Console.WriteLine($"   Original content:\n{Encoding.UTF8.GetString(originalContent)}");

            Console.WriteLine("3. Encrypting file...");
            var (ciphertext, encHeader) = encryptor.EncryptFile(testFile);
            Console.WriteLine($"   Ciphertext size: {ciphertext.Length} bytes");
            Console.WriteLine($"   Header size: {encHeader.Length} bytes");
            Console.WriteLine();

            var cipherFile = testFile + ".enc";
            File.WriteAllBytes(cipherFile, ciphertext);
            Console.WriteLine($"4. Saved ciphertext to: {cipherFile}");
            Console.WriteLine();

            Console.WriteLine("5. Decrypting file...");
            byte[] decrypted = decryptor.DecryptFile(cipherFile, encHeader);
            Console.WriteLine($"   Decrypted size: {decrypted.Length} bytes");
            Console.WriteLine($"   Decrypted content:\n{Encoding.UTF8.GetString(decrypted)}");

            if (!decrypted.SequenceEqual(originalContent))
                throw new ZuptException("Decryption failed!");
            Console.WriteLine();

            Console.WriteLine("6. Cleaned up temporary files");
            Console.WriteLine();
        }
        finally
        {
            if (Directory.Exists(tmpDir))
                Directory.Delete(tmpDir, recursive: true);
        }

        Console.WriteLine(new string('=', 60));
        Console.WriteLine("File encryption/decryption example passed!");
        Console.WriteLine(new string('=', 60));
        Console.WriteLine();
    }
}

File: examples_go/example_file.go
package main

import (
	"bytes"
	"errors"
	"fmt"
	"os"
	"path/filepath"
	"strings"
)

func runExampleFile() error {
	sep := strings.Repeat("=", 60)
	fmt.Println(sep)
	fmt.Println("libzupt - File Encryption/Decryption Example")
	fmt.Println(sep)
	fmt.Println()

	fmt.Println("1. Generating key pair...")
	keygen := NewKeyGenerator()
	keypair, err := keygen.GenerateKeyPair()
	if err != nil {
		return err
	}
	fmt.Println("   Key pair generated")
	fmt.Println()

	encryptor, err := NewEncryptor(keypair.PublicKey)
	if err != nil {
		return err
	}
	decryptor, err := NewDecryptor(keypair.SecretKey)
	if err != nil {
		return err
	}

	tmpDir, err := os.MkdirTemp("", "zupt_example_")
	if err != nil {
		return err
	}
	defer os.RemoveAll(tmpDir)

	testFile := filepath.Join(tmpDir, "example.txt")
	content := "This is a secret text file.\n" +
		"Line 2: Contains sensitive information.\n" +
		"Line 3: End of file.\n"
	original := []byte(content)
	if err := os.WriteFile(testFile, original, 0o600); err != nil {
		return err
	}

	fmt.Printf("2. Created test file: %s\n", testFile)
	fmt.Printf("   Original content:\n%s", original)
	fmt.Println()

	fmt.Println("3. Encrypting file...")
	ciphertext, encHeader, err := encryptor.EncryptFile(testFile)
	if err != nil {
		return err
	}
	fmt.Printf("   Ciphertext size: %d bytes\n", len(ciphertext))
	fmt.Printf("   Header size: %d bytes\n", len(encHeader))
	fmt.Println()

	cipherFile := testFile + ".enc"
	if err := os.WriteFile(cipherFile, ciphertext, 0o600); err != nil {
		return err
	}
	fmt.Printf("4. Saved ciphertext to: %s\n", cipherFile)
	fmt.Println()

	fmt.Println("5. Decrypting file...")
	decrypted, err := decryptor.DecryptFile(cipherFile, encHeader)
	if err != nil {
		return err
	}
	fmt.Printf("   Decrypted size: %d bytes\n", len(decrypted))
	fmt.Printf("   Decrypted content:\n%s", decrypted)
	fmt.Println()

	if !bytes.Equal(decrypted, original) {
		return errors.New("Decryption failed!")
	}

	fmt.Println("6. Cleaned up temporary files")
	fmt.Println()

	fmt.Println(sep)
	fmt.Println("File encryption/decryption example passed!")
	fmt.Println(sep)
	return nil
}
